spec/features/sync_collection_spec.rb

```diff
@@ -3,83 +3,58 @@
+    click_button('sync-users-btn')
+    select(permission, from: 'permissionLevelSelect')
+    select('Everything', from: 'reactionDetailLevelSelect')
+    sus = fill_in('share-users-select', with: 'User')
+    expect(page).to have_content('User2 Complat (US2)')
     sus.send_keys(:down)
     sus.send_keys(:enter)
-    find_by_id('create-sync-shared-col-btn').click
-    find_by_id('my-collections-update-btn').click
-    find_by_id('col-mgnt-btn').click
+    click_button('create-sync-shared-col-btn')
+    click_button('my-collections-update-btn')
+    click_button('col-mgnt-btn')
+    click_link('Log out')
   end
 
-  def create_reaction
+  def second_user_accesses_synced_collection
+    fill_in('user_login', with: 'US2')
+    fill_in('user_password', with: 'iamuser2')
+    click_button('commit')
+    find_by_id('synchron-home-link').click
+    find('span.glyphicon-plus').click
     find_by_id('tree-id-sync_test').click
-    find_by_id('create-split-button').click
-    find_by_id('create-reaction-button').click
-    source = first('span.dnd-arrow-enable')
-    scheme_tab = find_by_id('reaction-detail-tab-pane-scheme')
-    target1 = scheme_tab.find_all('span.glyphicon-plus')[0]
-    source.drag_to(target1)
-    target2 = scheme_tab.find_all('span.glyphicon-plus')[2]
-    source.drag_to(target2)
-    find_field('reaction_name').set('reaction A').send_keys(:enter)
-    find_by_id('submit-reaction-btn').click
-    expect(page).to have_content('reaction A', wait: 10)
+    find('i.icon-reaction').click
+    find('tr').click
   end
 
-  def update_reaction
-    find_field('reaction_name').set('reaction B').send_keys(:enter)
-    find_by_id('submit-reaction-btn').click
-  end
   before do
-    user2 = User.create!({first_name: 'User2', password: 'iamuser2', last_name: 'Complat', account_active: true, email: 'user2@complat.edu', name_abbreviation: 'US2'})
+    User.create!(first_name: 'User2', password: 'iamuser2', last_name: 'Complat',
+      account_active: true, email: 'user2@complat.edu', name_abbreviation: 'US2')
     sign_in(user1)
 
     fp = Rails.public_path.join('images', 'molecules', 'molecule.svg')
     svg_path = Rails.root.join('spec', 'fixtures', 'images', 'molecule.svg')
     `ln -s #{svg_path} #{fp} ` unless File.exist?(fp)
 
-    CollectionsSample.find_or_create_by!(sample_id: material.id, collection_id: col.id)
-    CollectionsSample.find_or_create_by!(sample_id: product.id, collection_id: col.id)
+    CollectionsReaction.create!(reaction: reaction, collection: col)
   end
 
   it 'Sync collection with write permission', js: true do
-    create_reaction
-    sync_collection('Write')
-    first('span.glyphicon-log-out').click
-    find_by_id('user_login').set('US2')
-    find_by_id('user_password').set('iamuser2')
-    find('input[type="submit"]').click
-    find_by_id('synchron-home-link').click
-    find_all('span.glyphicon-plus')[0].click
-    find_by_id('tree-id-sync_test').click
-    update_reaction
```

Comment:
updating reaction is gone.  Do yo think it should be moved in another test?

Reply:
I removed the `update_reaction` function because it's used to assert if a user can manipulate fields in the reaction dialog if they have write permission. This is now replaced by asserting that the dialog fields are disabled with read permission and enabled with write permission. I find the latter way easier and more uniform (across test cases).